src/client.ts

```diff
@@ -51,6 +53,30 @@ export class NuklaiCoreApiClient {
   }
 }
 
+export interface ActionResult {
+  asset_id: string;
+  dataset_address?: string;
+  dataset_contribution_id?: string;
+  marketplace_asset_address?: string;
+}
+
+export interface TransactionResult {
```

Comment:
I think as part of this, we should also return ActionInput where we return the parameters that went to execute each action. Also add the sponsor to this struct so we know who executed the action

Reply:
💯